prover/proof_producer/oracle_proof_producer.go

```diff
@@ -32,7 +37,13 @@ func NewOracleProducer(
 	proverPrivKey *ecdsa.PrivateKey,
 	taikoL2Address common.Address,
 	proofTimeTarget time.Duration,
+	protocolOracleProverAddress common.Address,
 ) (*OracleProducer, error) {
+	proverAddress := crypto.PubkeyToAddress(proverPrivKey.PublicKey)
+	if proverAddress != protocolOracleProverAddress {
+		return nil, errProtocolAddressMismatch
```

Comment:
i think in current protocol implementation, `msg.sender == protocolOracleProverAddress` is not required when submitting an oracle proof? ref: https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/contracts/L1/libs/LibProving.sol#L68